### PR TITLE
First attempt at managing the nfcore org with pulumi

### PR DESCRIPTION
**Pull request by bebosudo**
Hey Edmund, late for the hackaton, but I worked on issue #11 as we discussed.

This is a draft PR to manage the whole org with pulumi. It turns out it's easy enough once you get the feel of it.

I created a test organization `test-nfcore` and I used it to test the stack.
I'm not sure how to move things forwards now, and how to e.g. store the secret org token in GH to run this automatically once the org structure is changed? Just by using a GHA that sets up the token by taking it from a GH secret every time?
I used [this blogpost from pulumi](https://www.pulumi.com/blog/managing-github-with-pulumi/) to get started:
```
pulumi config set github:token "github_pat_***" --secret
pulumi config set github:owner test-nfcore
```

Which created a `Pulumi.dev.yaml` file (not commited in this PR). Any idea?

### CODE REVIEW

**Comment by edmundmiller** [comment]
Awesome! We also have `nf-core-tf` that we made for that purpose as well 😆 

I think the repo living here is the best pulumi cloud integration, things got weird when I tried to put the repo in the `nf-core-tf` org. 

As for storing the credentials, we can just make a token and store it in pulumi cloud (once they approve our team for open source) then whenever a PR is merged to master it will run is the idea.

Like I said on Gather town I think starting with Teams is a good way for us to safely get started, the worst case is that we accidentally delete a team and have to recreate it, not losing repos.

File: pulumi/github/teams/org.yaml
Comment (edmundmiller) [inline pulumi/github/teams/org.yaml:32]:
```suggestion
```

**Review by edmundmiller — COMMENTED** [review]
That all looks perfect to me, did it work?

**Comment by bebosudo** [comment]
Yep, it worked fine!
You should have received a notification to join one of the teams I created in the test org.

Also private and public repos were created successfully, but we can keep them out of our pulumi stack at the beginning, as you suggested.

Great to hear that we asked Pulumi to sponsor the project! Ping me once that is up and running

I just noticed that we use sub-teams within the nf-core org, e.g. [nf-core/infrastructure/ops](https://github.com/orgs/nf-core/teams/ops), so I'll need to make the script iterative.

**Comment by bebosudo** [comment]
BTW the Personal Access Token I created in the org needed the following permissions:

Organization permissions:
- Read and Write access to members

Repository permissions:
- Read access to metadata
- Read and Write access to administration

Repository Access
- Access to all current and future repositories owned by the organization [we can skip this for now if we decide not to use pulumi for repos]

**Comment by bebosudo** [comment]
I reworked the script to manage subteams.

I also added explicit dependencies between resources, otherwise sometimes the resource that associates a team with a repo (or a team with a member) was failing because one of the two resources wasn't created yet.

**Comment by edmundmiller** [comment]
Awesome work!

**Comment by edmundmiller** [comment]
To clarify, I think we can add a `nf-core-test` environment for testing purpose to pulumi. The issue was when I tried to use a repo under `nf-core-tf/pulumi` which pulumi cloud then thought that was our real org, and messed up the auth.

File: pulumi/github/teams/.gitignore
Comment (mashehu) [inline pulumi/github/teams/.gitignore:7]: are we sure we don't want to track the `requirements.txt`

File: pulumi/github/teams/requirements.txt
Comment (mashehu) [inline pulumi/github/teams/requirements.txt:3]: How about using [ruff](https://docs.astral.sh/ruff/) instead?

**Review by mashehu — APPROVED** [review]

File: pulumi/github/teams/.gitignore
Comment (bebosudo) [reply pulumi/github/teams/.gitignore:7]: The `requirements.txt` file was already part of this PR so it wouldn't get ignored in the future, but I added an explicit exception in the gitignore file.

File: pulumi/github/teams/requirements.txt
Comment (bebosudo) [reply pulumi/github/teams/requirements.txt]: Sure! It really looks great!
I updated the `requirements.txt` and ran `ruff` on the dir, but it was already properly formatted, so it produced no changes.

**Review by edmundmiller — APPROVED** [review]
Okay I've gotten this to deploy with Pulumi cloud. Had to just add the secret manually and couldn't get it to work with Pulumi ESC for whatever reason. 

Figured we can circle back around to that when we're comfortable using this for main nf-core org.

**Pull request merged**
